test/WebJobs.Script.Tests.Integration/ProxyEndToEndTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests
+{
+    public class ProxyEndToEndTests : EndToEndTestsBase<ProxyEndToEndTests.TestFixture>
+    {
+        public ProxyEndToEndTests(TestFixture fixture) : base(fixture)
+        {
+        }
+
+        [Fact]
+        public async Task Proxy_Mock_UrlTemplate()
+        {
+            HttpRequestMessage request = new HttpRequestMessage
+            {
+                RequestUri = new Uri(string.Format("http://localhost/mymockhttp")),
+                Method = HttpMethod.Get,
+            };
+
+            request.SetConfiguration(Fixture.RequestConfiguration);
+
+            Dictionary<string, object> arguments = new Dictionary<string, object>
+            {
+                { "req", request },
+            };
+            await Fixture.Host.CallAsync("localFunction_1002", arguments);
+
+            HttpResponseMessage response = (HttpResponseMessage)request.Properties[ScriptConstants.AzureFunctionsHttpResponseKey];
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(response.Headers.GetValues("myversion").ToArray()[0], "123");
+        }
```

Comment:
Currently the call to the local "Function" goes via the frontend. So its no differnt if the backend is local function or any other endpoint. Once we start using Function Executor we will have the scenario that would need special tests.

---
In reply to: [132565347](https://github.com/Azure/azure-webjobs-sdk-script/pull/1749#discussion_r132565347) [](ancestors = 132565347)